src/Reflection/ReflectionClass.php

```diff
@@ -331,6 +331,18 @@ public function hasProperty($name)
     }
 
     /**
+     * Protected API to add properties at runtime (used in ReflectionObject)
+     *
+     * @param ReflectionProperty $property
+     */
+    protected function addProperty(ReflectionProperty $property)
```

Comment:
Wait... you can add properties at runtime? WADDAFAK? 

If this is allowed, then it should happen in `ReflectionObject` internals, by overriding parent class properties rather than overwriting the `ReflectionClass` internal state.